File: 0392-is-subsequence/0392-is-subsequence.cpp
class Solution {
public:
    bool isSubsequence(string s, string t) {
        if (s.size() > t.size()) {
            return false;
        } 

        int cursor = 0;

        for (int i = 0; i < t.size(); i++) {
            if (t[i] == s[cursor]) {
                cursor++;
            }
        }

        return cursor == s.size();
    }
};

File: 0392-is-subsequence/0392-is-subsequence.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        ssz: int = len(s)
        tsz: int = len(t)

        if ssz > tsz:
            return False
        cur: int = 0

        for i in range(tsz):
            if cur == ssz:
                break
            if t[i] == s[cur]:
                cur+=1
        return cur == ssz